server/plugins/TriggerUGens.cpp

```diff
@@ -1459,9 +1462,14 @@ void Sweep_next_kk(Sweep* unit, int inNumSamples) {
     if (previn <= 0.f && curin > 0.f) {
         float frac = -previn / (curin - previn);
```

Comment:
I think this fraction calculation may be wrong.

Assume
```
previn = -0.25;
curin = 0.5;
```
The trigger happens at
```
-(-0.25) / (0.5 - -0.25)  = 1/3
```
i.e 1/3 of the way through the frame. So the resulting Sweep value should be `2/3 * rate`.

So this currently gives the "wrong side" of the increment ratio.

Reply:
In fact this bug was already caught in `Phasor`:
https://github.com/supercollider/supercollider/blame/7c6e99019d66f244ac871d75ddcaf900f6215c2e/server/plugins/TriggerUGens.cpp#L1614

So I suppose this `frac` calculation should  be checked throughout the whole source file (`Timer`, `ZeroCrossing`), and in UGens outside this source file. Something for another PR I suppose...

Reply:
Filed an Issue, #6883,  so I'll leave it up to you whether you'd like to include this fix or save that for another PR.

Reply:
It was caught in Phasor, but the fix is also still wrong.. it should be `1 - (-previn/(curin-previn))`